Gemfile

```diff
@@ -49,6 +49,12 @@ group :test do
   gem 'capybara', '>= 3.26'
   gem 'selenium-webdriver'
   # Easy installation and use of web drivers to run system tests with browsers
+  gem 'factory_bot'
```

Comment:
What does each library do?

Reply:
The FactoryBot gem has been used to create an article and a user in an easier way with the factories since without them it would be more complicated to do it and tend to redundancy.

The Shoulda matchers gem has been used to make a simpler syntax because it allows testing on one line and without this it would be more complicated to do so.

the faker gem is useful when you want to use fake data in the database for testing purposes.

The rails-controller-testing gem brings back assignsa that according to the documentation of this gem, these were eliminated since the rails 5 version, even so I have used it since I have had errors in the tests to the controller and according to the guides and documentation in internet this gem was a possible solution that for my case has worked